src/main/scala/org/constellation/infrastructure/endpoints/middlewares/PeerAuthMiddleware.scala

```diff
@@ -182,4 +271,15 @@ object PeerAuthMiddleware {
 
   private def getGenerator(privateKey: PrivateKey): Generator =
     GenericGenerator(KeyUtils.DefaultSignFunc, KeyUtils.provider, privateKey)
+
+  val constellationTokenHeader = CaseInsensitiveString("X-Constellation-Token")
```

Comment:
How about moving it to `constellation.session` along with `TokenService` and rename it to `SessionTokenService` and `X-Session-Token`?

Reply:
Sure! I still need to groom the code, but first I want it to work ;).